pandas/core/algorithms.py

```diff
@@ -2130,3 +2130,28 @@ def sort_mixed(values):
         np.putmask(new_codes, mask, na_sentinel)
 
     return ordered, ensure_platform_int(new_codes)
+
+
+def resort_union_after_inputs(union_values, lvals, rvals) -> np.ndarray:
+    """
+    Elements from union_values are resorted after the ranking of the first occurrence in
+    lvals and then rvals, if element is not in lvals. All occurrences of this element
+    are placed at the spot of the first occurrence of this element.
+
+    Parameters
+    ----------
+    union_values: np.array which is a sorted union of lvals and rvals
+    lvals: np.ndarray of the left values which is ordered in front.
+    rvals: np.ndarray of the right values ordered after lvals.
```

Comment:
could any of these be EAs?

Reply:
Not really, would work in principle, but numpy would cast result to object dtype. If this is ok, then yes, otherwise no.